### DISCUSSION TITLE
SolidWorks - Failed to load the process imported by 'CefSharp.Core.Runtime.dll'

### DISCUSSION THREAD

**Discussion by Jerry-Lnj** [cefsharp/CefSharp#4644] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

119.4.30.0

### Operating System

Windows 10

### Architecture

x64

### .Net Version

.net framework 4.6.2

### Implementation

WinForms

### Reproduction Steps

using CefSharp.WinForms;
using CefSharp;
using System.Runtime.InteropServices;
using System.Windows.Forms;

namespace PRM_solidworksPlug
{
    [ProgId(TaskpaneIntegration.SWTASKPANE_PROGID)]
    public partial class TaskpaneHostUI : UserControl
    {
        public TaskpaneHostUI()
        {
            InitializeComponent();
            InitBrowser("www.baidu.com");
        }
        public ChromiumWebBrowser browser;
        private void InitBrowser(string link)
        {
            CefSettings settings = new CefSettings();
            Cef.Initialize(settings, true);
            browser = new ChromiumWebBrowser(link);
            browser.Dock = DockStyle.Fill;
            this.Controls.Add(browser);
        }
    }
}

Using the code above will 100% reproduce the problem: 'Failed to load the process imported by 'CefSharp.Core.Runtime.dll'

### Expected behavior

Cefsharp webbrowser can load the website.

### Actual behavior

'Failed to load the process imported by 'CefSharp.Core.Runtime.dll' exception happened, if continue, blank on Cefsharp webbrowser. 

### Regression?

Background: I am trying to write a Solidworks plugin (dll file), function is pretty simple: Solidworks load a usercontrol, and only a cefsharp browser on the usercontrol. Code is tested and debugged on my own laptop. I need to deploy the program on other PC.

With the same setting (.net framework version, cefsharp version),  it worked properly on Solidworks 2017 (on my local PC and deployment on other PC). But when I try to deploy this on Solidworks 2021, error happened.

### Known Workarounds

I try the following steps:
1. Install all the required environment i can think about. Eg: .net framework 4.6.2 runtime, visual C++ distribution 2005-2022, not worked
2. Change other version of cefsharp, not worked
3. Write a standalone winform app, with cefsharp, this worked.

### Does this problem also occur in the CEF Sample Application

Not Tested

### Other information

![lQLPJxxC4P89kK_NA9vNB3awxNCdU1U5cj8FY8QskL7iAA_1910_987](https://github.com/cefsharp/CefSharp/assets/36815072/f530f984-1b19-4f93-b0c7-f1e556d4f3cf)
Here is the error page, sorry about the Chinese, but it means: Failed to load the process imported by 'CefSharp.Core.Runtime.dll'

![lQLPJw64y2oAYK_NA6zNByywxzD8UjsUn2cFY8QskL7iAg_1836_940](https://github.com/cefsharp/CefSharp/assets/36815072/b3cbdbac-7b3e-4f36-8e3c-2ac3f5169c2b)
Here is the plugin, showing blank.

![lQLPKd2K21Zbri_NAvjNBrawX-u49EY3oPMFY8QskL7iAQ_1718_760](https://github.com/cefsharp/CefSharp/assets/36815072/64844df2-3941-4c32-aa23-7291e0d199b5)
This is the source code, pretty simple. Error happen on Cef.Initialize();

**Comment by Jerry-Lnj** [comment]
![image](https://github.com/cefsharp/CefSharp/assets/36815072/fe5da32b-6b6f-4e4d-9bef-3ec3d7a497b9)
Found out a protential reason, SolidWorks integrated CEF since 2020. I am not sure if this SolidWorks CEF Sub Process will Interfere with my program (I am pretty sure that Cefsharp can only do one-time initialization)

**Reply by amaitland** [reply]
> Found out a protential reason, SolidWorks integrated CEF since 2020. I am not sure if this SolidWorks CEF Sub Process will Interfere with my program (I am pretty sure that Cefsharp can only do one-time initialization)

You can only load a single version of `CEF (Chromium)` per process. 

If Solidworks don't provide a browser control in their newer versions then you'll need to run `Chromium` out of process.

- There's an example of running `CefSharp` out of process at https://github.com/cefsharp/CefSharp.OutOfProcess
- Alternative is to switch to a different browser that by default runs the browser in a separate process.

**Comment by haroldcris** [comment]
same experience here.  Using Visual studio 2022 in my own created WinForm app. 
Still looking for solution

**Reply by Jerry-Lnj** [reply]
Are you trying to create a standalone app or built-in plugin. If standalone app, should be simple, will not have this confliction happens.

**Reply by amaitland** [reply]
> same experience here. Using Visual studio 2022 in my own created WinForm app.
> Still looking for solution

There are a number of different, unrelated reasons for the same error message. The most common is trying to run on an unsupported operating system (#4336)

If you are running on a supported OS and are still having problems then try https://github.com/cefsharp/CefSharp.MinimalExample which is a working example.

**Reply by haroldcris** [reply]
Thank you. You are right, turns out to be unsupported OS. (#4336)
